core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -63,11 +68,11 @@
   private final SerializingExecutor callExecutor;
   private final boolean unaryRequest;
   private final CallOptions callOptions;
-  private ClientStream stream;
+  private DelayedStream stream;
```

Comment:
`ClientCallImpl.start()` will skip `DelayedStream` when possible. If anything goes wrong during stream creation, `DelayedStream` will be created to handle the errors and possibly re-get the transport.